### PR TITLE
update(Metrics): Set Google Analytics userId if able.

### PR DESCRIPTION
**Pull request by alecklandgraf**
to: @milesj @stefhatcher @hayes @Alphy11 

## Description

Google Analytics isn't capturing users accurately. 
https://support.google.com/analytics/answer/3123666

## Motivation and Context

Allows setting the Google Analytics user ID in one shared place across all of Lunar's consumers.

## Testing

Updated tests.

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: types/global.d.ts [hunk 1]
```diff
@@ -78,6 +80,7 @@ declare namespace NodeJS {
     location: Location;
     navigator: Navigator;
     newrelic: NewRelic.Browser;
+    ga: Function | undefined;
```
Comment (milesj) [inline types/global.d.ts]: Are there actual types we can use here?

Comment (alecklandgraf) [reply types/global.d.ts]: yes actually, https://www.npmjs.com/package/@types/google.analytics

File: packages/metrics/src/index.ts [hunk 2]
```diff
@@ -28,6 +29,7 @@ class Metrics {
 
     this.bootstrapNewRelic();
     this.bootstrapSentry();
+    this.setGoogleAnalyticsUser();
```
Comment (milesj) [inline packages/metrics/src/index.ts]: `bootstrapGoogleAnalytics`

File: package-lock.json [hunk 3]
```diff
@@ -8606,12 +8613,14 @@
         "balanced-match": {
           "version": "1.0.0",
           "resolved": false,
-          "integrity": "sha1-ibTRmasr7kneFk6gK4nORi1xt2c="
+          "integrity": "sha1-ibTRmasr7kneFk6gK4nORi1xt2c=",
+          "optional": true
```
Comment (alecklandgraf) [inline package-lock.json]: @milesj are these `"optional": true` updates OK?

**Comment by alecklandgraf** [comment]
@milesj @stefhatcher PTAL

File: types/global.d.ts [hunk 4]
```diff
@@ -75,6 +75,7 @@ interface KeyboardEvent {
 declare namespace NodeJS {
   interface Global {
     __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: any;
+    ga: UniversalAnalytics.ga | any | undefined;
```
Comment (stefhatcher) [inline types/global.d.ts]: curious why the `any | undefined`? can make `ga` optional by adding a `?`:

```jsx
ga?: UniversalAnalytics.ga;
```

Comment (alecklandgraf) [reply types/global.d.ts]: will try that, I needed to be able to set `global.ga = undefined` in my tests

**Pull request merged**